c/include/ml-api-service.h

```diff
@@ -46,14 +47,182 @@ extern "C" {
+ * @since_tizen 9.0
+ * @remarks %http://tizen.org/privilege/mediastorage is needed if the configuration is relevant to media storage.
+ * @remarks %http://tizen.org/privilege/externalstorage is needed if the configuration is relevant to external storage.
+ * @remarks The @a handle should be released using ml_service_destroy().
+ * @param[in] config_path The absolute path to configuration file.
+ * @param[out] handle The handle of ml-service.
+ * @return @c 0 on success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #ML_ERROR_PERMISSION_DENIED The application does not have the privilege to access to the media storage or external storage.
+ * @retval #ML_ERROR_INVALID_PARAMETER The parameter is invalid.
+ * @retval #ML_ERROR_IO_ERROR Failed to parse the configuration file.
+ * @retval #ML_ERROR_STREAMS_PIPE Failed to open the model.
+ * @retval #ML_ERROR_OUT_OF_MEMORY Failed to allocate required memory.
+ *
+ * Here is an example of the usage:
+ * @code
+ *
+ * // Callback function for ml-service.
+ * // Note that the handle of tensors-data will be deallocated after the return and this is synchronously called.
+ * // Thus, if you need the data afterwards, copy the data to another buffer and return fast.
+ * // Do not spend too much time in the callback.
+ * static void
+ * _ml_service_cb_new_data (ml_service_h handle, const char *name, const ml_tensors_data_h data, void *user_data)
+ * {
+ *   void *_data;
+ *   size_t _size;
+ *
+ *   ml_tensors_data_get_tensor_data (data, 0, &_data, &_size);
+ *   // Handle output data.
+ * }
+ *
+ * // The path to the configuration file.
+ * const char config_path[] = "/path/to/application/configuration/my_application_config.conf";
+ *
+ * // Create ml-service for model inference from configuration.
+ * ml_service_h handle;
+ * ml_service_callbacks_s cb = { 0 };
+ *
+ * cb.new_data = _ml_service_cb_new_data;
+ *
+ * ml_service_new (config_path, &handle);
+ * ml_service_set_callback (handle, &cb, NULL);
+ *
+ * // Get input information and allocate input buffer.
+ * ml_tensors_info_h input_info;
+ * void *input_buffer;
+ * size_t input_size;
+
+ * ml_service_get_input_information (handle, NULL, &input_info);
+ *
+ * ml_tensors_info_get_tensor_size (input_info, 0, &input_size);
+ * input_buffer = malloc (input_size);
+ *
+ * // Create input data handle.
+ * ml_tensors_data_h input;
+ *
+ * ml_tensors_data_create (input_info, &input);
+ * ml_tensors_data_set_tensor_data (input, 0, input_buffer, input_size);
+ *
+ * // Push input data into ml-service and process the output in the callback.
+ * ml_service_add_input (handle, NULL, input);
+ *
+ * // Finally, release all handles and allocated memories.
+ * ml_tensors_info_destroy (input_info);
+ * ml_tensors_data_destroy (input);
+ * ml_service_destroy (handle);
+ * free (input_buffer);
+ *
+ * @endcode
+ */
+int ml_service_new (const char *config_path, ml_service_h *handle);
```

Comment:
How about adding `ml_service_start()` and `ml_service_stop` to handle the training offloading using ml_service?

Reply:
I added new function _start()/_stop() for future work. Please see the changes again.